maraboupy/MarabouNetworkNNet.py

```diff
@@ -49,13 +54,111 @@ def __init__ (self, filename, normalize=False):
             weights          (list of list of lists) Outer index corresponds to layer
                                 number.
             biases           (list of lists) Outer index corresponds to layer number.
+
+         others:
+            inputVars
+            b_variables
+            f_variables
+            outputVars
+            numVars
+
+        Attributes from parent (MarabouNetwork)
+
+            numVars
+            equList = []
+            reluList = []
+            maxList = []
+            varsParticipatingInConstraints = set()
+            lowerBounds = dict()
+            upperBounds = dict()
+            inputVars = []
+            outputVars = np.array([])
         """
         super().__init__()
 
         self.normalize = normalize
+
+        if not filename:
+            self.numLayers = 0
+            self.layerSizes = []
+            self.inputSize = 0
+            self.outputSize = 0
+            self.maxLayersize = 0
+            self.inputMinimums = []
+            self.inputMaximums = []
+            self.inputMeans = []
+            self.inputRanges = []
+            self.outputMean = 0
+            self.outputRange = 0
+            self.weights = []
+            self.biases = []
+
+            # self.nlr = None
+            return
+
         # Read the file and load values
         self.read_nnet(filename)
 
+        self.computeNetworkAttributes()
+
+    def clearNetwork(self):
+        # call clear() from parent
+        self.clear()
+
+        self.numLayers = 0
+        self.layerSizes = 0
```

Comment:
self.layerSizes should default to an empty list

Reply:
My bad. Thanks!